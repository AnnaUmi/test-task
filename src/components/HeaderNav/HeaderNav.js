import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { HeaderContainer, HeaderLogo, NavMenu, NavItem } from './HeaderNav.styles';

const renderNav = (nav, currRoute) =>
  nav.map(({ route, id, label }) => (
    <Link href={route} key={id}>
      <NavItem isActive={currRoute === route}>
        <div>{label}</div>
      </NavItem>
    </Link>
  ));

const HeaderNav = ({ currMainRoute, routes }) => {
  const logo = routes[0];
  const nav = routes.slice(1, routes.length);

  return (
    <HeaderContainer>
      <Link href={logo.route}>
        <HeaderLogo>{logo.label}</HeaderLogo>
      </Link>
      <NavMenu>{renderNav(nav, currMainRoute)}</NavMenu>
    </HeaderContainer>
  );
};

HeaderNav.propTypes = {
  currMainRoute: PropTypes.string.isRequired,
  routes: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default HeaderNav;
