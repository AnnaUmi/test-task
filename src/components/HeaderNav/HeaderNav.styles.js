import styled from 'styled-components';

export const HeaderContainer = styled.div`
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  align-items: center;
`;

export const HeaderLogo = styled.div`
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
`;

export const NavMenu = styled.div`
  margin-left: 40px;
  display: flex;
`;

export const NavItem = styled.div`
  margin-left: 40px;
  cursor: pointer;
  display: flex;
  flex-direction: column;

  &:first-child {
    margin-left: 0;
  }

  :after {
    content: '';
    display: inline-block;
    width: 100%;
    height: 2px;
    margin-top: 2px;
    background-color: ${({ isActive }) => (isActive ? 'blue' : 'rgba(0,0,0,0)')};
  }

  :hover {
    :after {
      content: '';
      display: inline-block;
      width: 100%;
      height: 2px;
      visibility: 'visible';
      background-color: ${({ isActive }) => (isActive ? 'blue' : '#696969')};
    }
  }
`;
