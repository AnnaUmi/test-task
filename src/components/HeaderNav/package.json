{
  "name": "header-nav",
  "version": "0.0.0",
  "private": true,
  "main": "./HeaderNav.js"
}
