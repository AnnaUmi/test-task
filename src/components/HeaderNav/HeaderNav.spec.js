import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import HeaderNav from './HeaderNav';

const mockRoutes = [
  {
    route: '/',
    label: 'PRISM Logo',
    id: '/',
  },
  {
    route: '/brand',
    label: 'Brand',
    id: '/brand',
  },
  {
    route: '/product',
    label: 'Product',
    id: '/product',
  },
  {
    route: '/marketing',
    label: 'Marketing',
    id: '/marketing',
  },
  {
    route: '/components',
    label: 'Components',
    id: '/components',
    subRoutes: [
      {
        route: '/components/buttons',
        label: 'Buttons',
        id: 'components/buttons',
      },
      {
        route: '/components/icons',
        label: 'Icons',
        id: 'components/icons',
      },
      {
        route: '/components/grid',
        label: 'Grid',
        id: 'components/grid',
      },
    ],
  },
];

describe('HeaderNav', () => {
  it('should match snapshot route is root', () => {
    const tree = renderer.create(<HeaderNav routes={mockRoutes} currMainRoute="/" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot route is brand', () => {
    const tree = renderer.create(<HeaderNav routes={mockRoutes} currMainRoute="/brand" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot route is product', () => {
    const tree = renderer
      .create(<HeaderNav routes={mockRoutes} currMainRoute="/product" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot route is marketing', () => {
    const tree = renderer
      .create(<HeaderNav routes={mockRoutes} currMainRoute="/marketing" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot route is components', () => {
    const tree = renderer
      .create(<HeaderNav routes={mockRoutes} currMainRoute="/components" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot route is buttons inside components', () => {
    const tree = renderer
      .create(<HeaderNav routes={mockRoutes} currMainRoute="/components/buttons" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
