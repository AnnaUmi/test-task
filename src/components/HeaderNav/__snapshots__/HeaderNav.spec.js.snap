// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav should match snapshot route is brand 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
}

.c2 {
  margin-left: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4:first-child {
  margin-left: 0;
}

.c4:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: rgba(0,0,0,0);
}

.c4:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: #696969;
}

.c3 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3:first-child {
  margin-left: 0;
}

.c3:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: blue;
}

.c3:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: blue;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    PRISM Logo
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Brand
      </div>
    </div>
    <div
      className="c4"
      onClick={[Function]}
    >
      <div>
        Product
      </div>
    </div>
    <div
      className="c4"
      onClick={[Function]}
    >
      <div>
        Marketing
      </div>
    </div>
    <div
      className="c4"
      onClick={[Function]}
    >
      <div>
        Components
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderNav should match snapshot route is buttons inside components 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
}

.c2 {
  margin-left: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3:first-child {
  margin-left: 0;
}

.c3:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: rgba(0,0,0,0);
}

.c3:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: #696969;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    PRISM Logo
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Brand
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Product
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Marketing
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Components
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderNav should match snapshot route is components 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
}

.c2 {
  margin-left: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3:first-child {
  margin-left: 0;
}

.c3:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: rgba(0,0,0,0);
}

.c3:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: #696969;
}

.c4 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4:first-child {
  margin-left: 0;
}

.c4:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: blue;
}

.c4:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: blue;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    PRISM Logo
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Brand
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Product
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Marketing
      </div>
    </div>
    <div
      className="c4"
      onClick={[Function]}
    >
      <div>
        Components
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderNav should match snapshot route is marketing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
}

.c2 {
  margin-left: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3:first-child {
  margin-left: 0;
}

.c3:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: rgba(0,0,0,0);
}

.c3:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: #696969;
}

.c4 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4:first-child {
  margin-left: 0;
}

.c4:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: blue;
}

.c4:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: blue;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    PRISM Logo
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Brand
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Product
      </div>
    </div>
    <div
      className="c4"
      onClick={[Function]}
    >
      <div>
        Marketing
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Components
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderNav should match snapshot route is product 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
}

.c2 {
  margin-left: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3:first-child {
  margin-left: 0;
}

.c3:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: rgba(0,0,0,0);
}

.c3:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: #696969;
}

.c4 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4:first-child {
  margin-left: 0;
}

.c4:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: blue;
}

.c4:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: blue;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    PRISM Logo
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Brand
      </div>
    </div>
    <div
      className="c4"
      onClick={[Function]}
    >
      <div>
        Product
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Marketing
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Components
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderNav should match snapshot route is root 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #d3d3d3;
  padding: 0 20px;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 1px dashed #696969;
  padding: 5px 0;
  min-width: 120px;
  text-align: center;
  cursor: pointer;
}

.c2 {
  margin-left: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-left: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3:first-child {
  margin-left: 0;
}

.c3:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  margin-top: 2px;
  background-color: rgba(0,0,0,0);
}

.c3:hover:after {
  content: '';
  display: inline-block;
  width: 100%;
  height: 2px;
  visibility: 'visible';
  background-color: #696969;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    PRISM Logo
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Brand
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Product
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Marketing
      </div>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <div>
        Components
      </div>
    </div>
  </div>
</div>
`;
